File: javascript/examples/basic_usage.js
import { BanglaAI } from "../src/index.js";

const ai = new BanglaAI({ apiKey: process.env.OPENAI_API_KEY });

const text =
  "বাংলাদেশ দক্ষিণ এশিয়ার একটি দেশ। এর রাজধানী ঢাকা। " +
  "বাংলাদেশের মোট জনসংখ্যা প্রায় ১৭ কোটি। " +
  "দেশটি ১৯৭১ সালে স্বাধীনতা লাভ করে।";

// Summarization
console.log("=== Summarization ===");
console.log(await ai.summarize.call(text, { maxSentences: 2 }));
console.log();

// Question Answering
const context =
  "রবীন্দ্রনাথ ঠাকুর ১৮৬১ সালে কলকাতায় জন্মগ্রহণ করেন। " +
  "তিনি ১৯১৩ সালে সাহিত্যে নোবেল পুরস্কার লাভ করেন।";

console.log("=== Question Answering ===");
console.log(await ai.qa.call(context, "রবীন্দ্রনাথ কোথায় জন্মগ্রহণ করেন?"));
console.log();

// Sentiment Analysis
console.log("=== Sentiment Analysis ===");
const result = await ai.sentiment.call(
  "আজকের দিনটি অসাধারণ ছিল! সব কিছু ভালো হয়েছে।"
);
console.log(`Label: ${result.label}`);
console.log(`Score: ${result.score.toFixed(2)}`);
console.log(`Explanation: ${result.explanation}`);

File: python/examples/basic_usage.py
"""bangla-ai usage examples."""

import os

# ── 1. Text utilities — zero dependencies ──────────────────────────────────────
from bangla_ai import text

raw = "বাংলাদেশ   দক্ষিণ এশিয়ার একটি দেশ।  এর রাজধানী ঢাকা।"

normalized = text.normalize(raw)
tokens = text.tokenize(normalized)
sentences = text.sent_tokenize(normalized)
filtered = text.remove_stopwords(tokens)
stems = text.stem_tokens(filtered)

print("=== Text Utilities ===")
print("Normalized :", normalized)
print("Tokens     :", tokens)
print("Sentences  :", sentences)
print("Filtered   :", filtered)
print("Stems      :", stems)
print()

# ── 2. HuggingFace backend (free) ──────────────────────────────────────────────
# Uncomment after: pip install bangla-ai[transformers]
#
# from bangla_ai import BanglaAI
# ai = BanglaAI()  # downloads models on first run
# ...

# ── 3. OpenAI backend ──────────────────────────────────────────────────────────
from bangla_ai import BanglaAI

ai = BanglaAI(backend="openai", api_key=os.environ["OPENAI_API_KEY"])

passage = (
    "বাংলাদেশ দক্ষিণ এশিয়ার একটি দেশ। এর রাজধানী ঢাকা। "
    "দেশটি ১৯৭১ সালে স্বাধীনতা লাভ করে।"
)

print("=== Summarization ===")
print(ai.summarize(passage, max_sentences=1))
print()

print("=== Question Answering ===")
print(ai.qa(passage, "বাংলাদেশের রাজধানী কোথায়?"))
print()

print("=== Sentiment Analysis ===")
r = ai.sentiment("আজকের দিনটি অসাধারণ ছিল!")
print(f"Label: {r['label']}  Score: {r['score']:.2f}")
print(f"Explanation: {r['explanation']}")
